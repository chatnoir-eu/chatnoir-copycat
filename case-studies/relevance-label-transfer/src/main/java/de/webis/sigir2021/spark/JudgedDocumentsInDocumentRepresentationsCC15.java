package de.webis.sigir2021.spark;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;
import java.util.stream.Collectors;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

import de.webis.trec_ndd.trec_collections.SharedTask;
import de.webis.trec_ndd.trec_collections.SharedTask.TrecSharedTask;

public class JudgedDocumentsInDocumentRepresentationsCC15 {

	public static void main(String[] args) {
		try (JavaSparkContext context = context()) {
			Map<String, String> docIdToUrl = JudgedDocumentsInDocumentRepresentations.docIdToUrl(context, judgedCW09AndCW12Documents(), 25);
			JavaRDD<String> sourceDocs = context.textFile("sigir2021/{cw09,cw12}-repartitioned");
			
			Map<String, String> secondDocToUrl = JudgedDocumentsInDocumentRepresentations.docIdToUrl(sourceDocs, docIdToUrl);
			Map<String, List<String>> urlToDocId = new LinkedHashMap<>();
			
			for(Map.Entry<String, List<String>> tmp : JudgedDocumentsInDocumentRepresentations.urltoDocId(docIdToUrl).entrySet()) {
				if(!urlToDocId.containsKey(tmp.getKey())) {
					urlToDocId.put(tmp.getKey(), new ArrayList<>());
				}
				
				urlToDocId.get(tmp.getKey()).addAll(tmp.getValue());
			}
			
			for(Map.Entry<String, List<String>> tmp : JudgedDocumentsInDocumentRepresentations.urltoDocId(secondDocToUrl).entrySet()) {
				if(!urlToDocId.containsKey(tmp.getKey())) {
					urlToDocId.put(tmp.getKey(), new ArrayList<>());
				}
				
				urlToDocId.get(tmp.getKey()).addAll(tmp.getValue());
			}
			
			JavaRDD<String> cc15DocumentRepresentations = context.textFile("sigir2021/cc-2015-11-repartitioned");
			JudgedDocumentsInDocumentRepresentations.urlSearch2(urlToDocId, cc15DocumentRepresentations)
				.saveAsTextFile("sigir2021/url-judgments-from-cw09-or-cw12-to-cc-2015-11");
		}
	}
	
	public static List<String> judgedCW09AndCW12Documents() {
		Set<String> ret = new HashSet<>();
		
		for(SharedTask task: new SharedTask[] {TrecSharedTask.WEB_2009, TrecSharedTask.WEB_2010, TrecSharedTask.WEB_2011, TrecSharedTask.WEB_2012, TrecSharedTask.WEB_2013, TrecSharedTask.WEB_2014}) {
			for(String topic: task.documentJudgments().topics()) {
				ret.addAll(task.documentJudgments().getIrrelevantDocuments(topic));
				ret.addAll(task.documentJudgments().getRelevantDocuments(topic));
			}
		}

		return ret.stream().sorted().collect(Collectors.toList());
	}
	
	private static JavaSparkContext context() {
		SparkConf conf = new SparkConf(true);
		conf.setAppName("JudgedDocumentsInDocumentRepresentations");

		return new JavaSparkContext(conf);
	}
}
