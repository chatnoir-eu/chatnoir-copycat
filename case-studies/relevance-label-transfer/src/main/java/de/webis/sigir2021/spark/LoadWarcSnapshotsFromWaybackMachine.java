package de.webis.sigir2021.spark;

import java.io.Serializable;
import java.net.URL;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

import org.apache.commons.lang.StringUtils;
import org.apache.hadoop.io.LongWritable;
import org.apache.hadoop.io.compress.GzipCodec;
import org.apache.htrace.fasterxml.jackson.databind.ObjectMapper;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;

import com.google.common.collect.ImmutableList;

import de.webis.chatnoir2.mapfile_generator.inputformats.CommonCrawlInputFormat;
import de.webis.chatnoir2.mapfile_generator.warc.WarcRecord;
import de.webis.sigir2021.wayback_machine.JudgedDocumentsWarcReader;
import de.webis.trec_ndd.trec_collections.CollectionDocument;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.SneakyThrows;
import scala.Tuple2;

public class LoadWarcSnapshotsFromWaybackMachine {

	public static void main(String[] args) {
		try(JavaSparkContext context = context()) {
			JavaPairRDD<String, List<CollectionDocument>> records = topicToWaybackDocumentsInCW12CrawlingTime(
				context,
				"file:///mnt/ceph/storage/data-in-progress/kibi9872/sigir2021/trec-judgments-in-wayback-machine/{redirects,snapshots}/*/*/*/*.warc.gz"
			);
			
			records.map(i -> new TmpContent(i)).map(i -> i.toString())
				.repartition(10)
				.saveAsTextFile("sigir2021/trec-judgments-in-wayback-machine/redirects-and-snapshots-as-collection-documents.jsonl", GzipCodec.class);
		}
	}
	
	private static JavaSparkContext context() {
		SparkConf conf = new SparkConf(true);
		conf.setAppName("LoadWarcSnapshotsFromWaybackMachine");

		return new JavaSparkContext(conf);
	}
	
	@Data
	@NoArgsConstructor
	@AllArgsConstructor
	@SuppressWarnings("serial")
	public static class TmpContent implements Serializable {
		private String trecDocumentId;
		private List<CollectionDocument> documents;
		
		@Override
		@SneakyThrows
		public String toString() {
			return new ObjectMapper().writeValueAsString(this);
		}

		public TmpContent(Tuple2<String, List<CollectionDocument>> i) {
			this(i._1(), i._2());
		}
	}
	
	public static JavaPairRDD<String, List<CollectionDocument>> topicToWaybackDocumentsInCW12CrawlingTime(JavaSparkContext jsc, String input) {
		JavaPairRDD<String, List<CollectionDocument>> ret = topicToWaybackDocumentsAll(jsc, input);
		ret = ret.mapToPair(i -> keepOnlyRecordsWithinCw12(i));
		
		return ret.filter(i -> i != null && i._1() != null && i._2() != null && !i._2().isEmpty());
	}

	private static Tuple2<String, List<CollectionDocument>> keepOnlyRecordsWithinCw12(Tuple2<String, List<CollectionDocument>> i) {
		List<CollectionDocument> docs = new ArrayList<>(i._2.stream().filter(doc -> keepOnlyRecordsWithinCw12(doc)).collect(Collectors.toList()));
		
		return new Tuple2<>(i._1, docs);
	}
	
	@SneakyThrows
	static boolean keepOnlyRecordsWithinCw12(CollectionDocument doc) {
		try {
			if(doc.getCrawlingTimestamp() == null) {
				return true;
			}
			
			if(doc.getCrawlingTimestamp().contains(" 2009") || doc.getCrawlingTimestamp().contains(" 2010")) {
				return false;
			}
			
			Date crawlingDate = parseDate(doc);
			return JudgedDocumentsWarcReader.responseInCw12CrawlingTime(crawlingDate);
		} catch(Exception e) {
			throw new RuntimeException("Could not parse: '" + doc.getCrawlingTimestamp() + "'.");
		}
	}
	
	@SneakyThrows
	private static Date parseDate(CollectionDocument doc) {
		SimpleDateFormat sdf = new SimpleDateFormat("dd MMM yyyy hh:mm:ss zzz");
		try {
			return sdf.parse(StringUtils.substringAfter(doc.getCrawlingTimestamp().replaceAll("\\s+", " "), ","));
		} catch (Exception e) {
			sdf = new SimpleDateFormat("dd-MMM-yyyy hh:mm:ss zzz");
			return sdf.parse(StringUtils.substringAfter(doc.getCrawlingTimestamp().replaceAll("\\s+", " "), ","));
		}
	}

	static JavaPairRDD<String, List<CollectionDocument>> topicToWaybackDocumentsAll(JavaSparkContext jsc, String input) {
		JavaPairRDD<LongWritable, WarcRecord> records = records(jsc, input);
		JavaPairRDD<String, CollectionDocument> topicToDoc = records.mapToPair(i -> transformToCollectionDocument(i))
				.filter(i -> i != null && i._1() != null && i._2() != null);
		
		return topicToDoc.groupByKey().mapToPair(i -> joinLists(i));
	}
	
	private static Tuple2<String, List<CollectionDocument>> joinLists(Tuple2<String, Iterable<CollectionDocument>> i) {
		return new Tuple2<>(i._1(), new ArrayList<>(ImmutableList.copyOf(i._2())));
	}

	private static Tuple2<String, CollectionDocument> transformToCollectionDocument(Tuple2<LongWritable, WarcRecord> i) {
		return transformToCollectionDocument(i._2());
	}
	
	@SneakyThrows
	public static Tuple2<String, CollectionDocument> transformToCollectionDocument(WarcRecord record) {
		if (record == null) {
			return null;
		}

		Map<String, String> header = lowercasedHeaders(record);
		String contentBody = record.getContent();

		if (contentBody.getBytes().length > 1024 * 1024 || !isWarcResponse(record)) {
			// ignore large files and non-responses
			return null;
		}
		
		try {
			return transformToCollectionDocument(header, Jsoup.parse(contentBody));
		} catch (Exception e) {
			return null;
		}
	}
	
	@SneakyThrows
	private static Tuple2<String, CollectionDocument> transformToCollectionDocument(Map<String, String> header, Document doc) {
		String id = header.get("warc-trec-id");
		if (id == null || id.isEmpty()) {
			id = header.get("warc-record-id");
		}

		String targetUri = header.get("warc-target-uri");

		removeAllWaybackContent(doc);
		CollectionDocument ret = CollectionDocument.collectionDocument(doc.text(), id);
		try {
			ret.setUrl(new URL(targetUri));
		} catch (Exception e) {}
		ret.setCrawlingTimestamp(header.get("x-archive-orig-date"));

		return new Tuple2<>(header.get("trec_target_id"), ret);
	}
	
    public static String removeAllWaybackContent(String html) {
        if (null == html || html.trim().isEmpty()) {
            return "";
        }
        
        Document doc = Jsoup.parse(html);
        removeAllWaybackContent(doc);
        
        return doc.toString();
    }
    
	private static void removeAllWaybackContent(Document doc) {
		for(Element e: doc.select("div[id=wm-ipp-base]")) {
			e.remove();
		}
	}
	
	public static Map<String, String> lowercasedHeaders(WarcRecord record) {
		return lowercasedHeaders(record.getHeader().getHeaderMetadata());
	}
	
	public static Map<String, String> lowercasedHeaders(Map<String, String> headers) {
		return new HashMap<>(headers.entrySet().stream()
				.filter(i -> i != null && i.getKey() != null && i.getValue() != null)
				.collect(Collectors.toMap(i -> i.getKey().trim().toLowerCase(), i -> i.getValue())));
	}
	
	public static boolean isWarcResponse(WarcRecord record) {
		return record != null && record.getRecordType() != null && "response".equalsIgnoreCase(record.getRecordType().trim());
	}

	private static JavaPairRDD<LongWritable, WarcRecord> records(JavaSparkContext sc, String path) {
		return sc.newAPIHadoopFile(path, CommonCrawlInputFormat.class, LongWritable.class, WarcRecord.class, sc.hadoopConfiguration());
	}
}
