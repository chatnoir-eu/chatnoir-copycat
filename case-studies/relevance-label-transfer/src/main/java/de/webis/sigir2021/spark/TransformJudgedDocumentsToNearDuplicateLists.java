package de.webis.sigir2021.spark;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.json.JSONArray;
import org.json.JSONObject;

public class TransformJudgedDocumentsToNearDuplicateLists {

	public static void main(String[] args) {
		try (JavaSparkContext context = context()) {
			JavaRDD<String> input = context.textFile("sigir2021/url-judgments-from-cw09-to-cw12");
			
			toDuplicatePairs(input)
				.saveAsTextFile("sigir2021/url-judgments-from-cw09-to-cw12-with-similarity");
		}
	}
	
	private static JavaSparkContext context() {
		SparkConf conf = new SparkConf(true);
		conf.setAppName("JudgedDocumentsInDocumentRepresentations");

		return new JavaSparkContext(conf);
	}
	
	public static JavaRDD<String> toDuplicatePairs(JavaRDD<String> input) {
		return input.flatMap(i -> toDuplicatePairs(i));
	}
	
	private static Iterator<String> toDuplicatePairs(String src) {
		List<String> ret = new ArrayList<>();
		JSONObject parsedJson = new JSONObject(src);
		String sourceId = parsedJson.getString("sourceId");
		JSONArray targetIds = parsedJson.getJSONArray("targetIds");

		for(int i=0; i<targetIds.length(); i++) {
			ret.add("{\"firstId\":\"" + sourceId + "\",\"secondId\":\"" + targetIds.getString(i) + "\",\"hemmingDistance\":-1}");
		}
		
		return ret.iterator();
	}
}
