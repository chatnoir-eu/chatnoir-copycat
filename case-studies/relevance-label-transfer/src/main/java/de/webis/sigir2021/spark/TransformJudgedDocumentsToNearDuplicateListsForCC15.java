package de.webis.sigir2021.spark;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

public class TransformJudgedDocumentsToNearDuplicateListsForCC15 {
	public static void main(String[] args) {
		try (JavaSparkContext context = context()) {
			JavaRDD<String> input = context.textFile("sigir2021/url-judgments-from-cw09-or-cw12-to-cc-2015-11");
			
			TransformJudgedDocumentsToNearDuplicateLists.toDuplicatePairs(input)
				.saveAsTextFile("sigir2021/url-judgments-from-cw09-or-cw12-to-cc-2015-11-with-similarity");
		}
	}
	
	private static JavaSparkContext context() {
		SparkConf conf = new SparkConf(true);
		conf.setAppName("TransformJudgedDocumentsToNearDuplicateListsForCC15");

		return new JavaSparkContext(conf);
	}
}
