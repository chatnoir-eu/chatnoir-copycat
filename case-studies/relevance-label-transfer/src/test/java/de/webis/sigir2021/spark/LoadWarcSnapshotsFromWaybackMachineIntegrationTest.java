package de.webis.sigir2021.spark;

import java.io.Serializable;
import java.util.Arrays;
import java.util.List;

import org.apache.hadoop.io.compress.GzipCodec;
import org.apache.htrace.fasterxml.jackson.databind.ObjectMapper;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.approvaltests.Approvals;
import org.junit.Assert;
import org.junit.Test;

import de.webis.sigir2021.spark.LoadWarcSnapshotsFromWaybackMachine;
import de.webis.sigir2021.spark.LoadWarcSnapshotsFromWaybackMachine.TmpContent;
import de.webis.trec_ndd.trec_collections.CollectionDocument;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.SneakyThrows;
import scala.Tuple2;

public class LoadWarcSnapshotsFromWaybackMachineIntegrationTest extends SparkIntegrationTestBase {
	@Test
	public void testParsingOfStuff() {
		String input = "src/test/resources/sample-input/snapshot-wayback-topic-02.warc.gz";
		JavaPairRDD<String, List<CollectionDocument>> records = LoadWarcSnapshotsFromWaybackMachine.topicToWaybackDocumentsAll(jsc(), input);

		records = records.filter(i -> Arrays.asList("clueweb09-en0000-40-26134", "clueweb09-en0003-39-09826", "clueweb09-enwp00-51-09934", "clueweb09-enwp00-78-20657").contains(i._1()));
		Approvals.verifyAsJson(records.collect());
	}
	
	@Test
	public void testParsingAndFiltering1() {
		String input = "src/test/resources/sample-input/snapshot-wayback-topic-02.warc.gz";
		JavaPairRDD<String, List<CollectionDocument>> records = LoadWarcSnapshotsFromWaybackMachine.topicToWaybackDocumentsInCW12CrawlingTime(jsc(), input);
		// ersten zwei fliegen raus, letzten zwei bleiben drin
		records = records.filter(i -> Arrays.asList("clueweb09-en0000-40-26134", "clueweb09-en0003-39-09826", "clueweb09-enwp00-51-09934", "clueweb09-enwp00-78-20657").contains(i._1()));
		Approvals.verifyAsJson(records.collect());
	}
	
	@Test
	public void testParsingOfStuff2() {
		String input = "src/test/resources/sample-input/snapshot-wayback-topic-02.warc.gz";
		JavaPairRDD<String, List<CollectionDocument>> records = LoadWarcSnapshotsFromWaybackMachine.topicToWaybackDocumentsAll(jsc(), input);

		records = records.filter(i -> Arrays.asList("clueweb09-en0006-06-41866", "clueweb09-enwp02-04-08736").contains(i._1()));
		Approvals.verifyAsJson(records.collect());
	}
	
	@Test
	public void testParsingAndFiltering2() {
		String input = "src/test/resources/sample-input/snapshot-wayback-topic-02.warc.gz";
		JavaPairRDD<String, List<CollectionDocument>> records = LoadWarcSnapshotsFromWaybackMachine.topicToWaybackDocumentsInCW12CrawlingTime(jsc(), input);
		// clueweb09-en0006-06-41866 raus, clueweb09-enwp02-04-08736 rein
		records = records.filter(i -> Arrays.asList("clueweb09-en0006-06-41866", "clueweb09-enwp02-04-08736").contains(i._1()));
		Approvals.verifyAsJson(records.collect());
	}
	
	@Test
	public void testParsingOfTimestamp() {
		CollectionDocument doc = doc("Mon, 10-Oct-2011 10:41:30 GMT");
		
		boolean actual = LoadWarcSnapshotsFromWaybackMachine.keepOnlyRecordsWithinCw12(doc);
		
		Assert.assertFalse(actual);
	}
	
	@Test
	public void testParsingOfTimestamp2() {
		CollectionDocument doc = doc("Sun Nov 08 19:45:59 GMT 2009");
		
		boolean actual = LoadWarcSnapshotsFromWaybackMachine.keepOnlyRecordsWithinCw12(doc);
		
		Assert.assertFalse(actual);
	}
	
//	
//	@Test
//	public void createBigJsonFile() {
//		String input = allWarcs();
//		JavaPairRDD<String, List<CollectionDocument>> records = LoadWarcSnapshotsFromWaybackMachine.topicToWaybackDocumentsInCW12CrawlingTime(jsc(), input);
//		
//		
//		Approvals.verifyAsJson(records.map(i -> new TmpContent(i)).take(1));
//		records.map(i -> new TmpContent(i)).map(i -> i.toString())
//			.repartition(10)
//			.saveAsTextFile("/mnt/ceph/storage/data-in-progress/kibi9872/sigir2021/trec-judgments-in-wayback-machine/redirects-and-snapshots-as-collection-documents.jsonl", GzipCodec.class);
//	}
	
	private static CollectionDocument doc(String date) {
		CollectionDocument ret = new CollectionDocument();
		ret.setCrawlingTimestamp(date);
		
		return ret;
	}
	
	private static final String allWarcs() {
		return "/mnt/ceph/storage/data-in-progress/kibi9872/sigir2021/trec-judgments-in-wayback-machine/{redirects,snapshots}/*/*/*/*.warc.gz";
	}
}
