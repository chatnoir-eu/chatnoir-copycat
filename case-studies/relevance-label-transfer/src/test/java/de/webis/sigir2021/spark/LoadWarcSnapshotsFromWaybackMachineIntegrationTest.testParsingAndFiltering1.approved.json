[
  {
    "_1": "clueweb09-enwp00-51-09934",
    "_2": [
      {
        "id": "\u003curn:uuid:ec9284c3-47c3-451e-b415-b8ff5c91a869\u003e",
        "content": "Chukchansi Gold Resort \u0026 Casino - Wikipedia, the free encyclopedia Chukchansi Gold Resort \u0026 Casino From Wikipedia, the free encyclopedia Jump to: navigation, search Chukchansi Gold Resort \u0026 Casino Address 711 Lucky Lane Coarsegold, California 93614 No. of rooms 192 Total gaming space 56,000 sq ft (5,200 m2) Notable restaurants California Market Buffet Vintage Steakhouse Deuces Diner Noodle Bar Asian Bistro Goldfields Cafe The Bakery Firehouse Blues Casino type Land-based Owner Picayune Rancheria of Chukchansi Indians Website http://www.chukchansigold.com/ Chukchansi Gold Resort \u0026 Casino is located just off of State Route 41 in Coarsegold, California, between Fresno and Yosemite National Park. It is owned and operated by the Picayune Rancheria of Chukchansi Indians.[1] The 56,000-square-foot (5,200 m2) casino has 1,800 slot machines, as well as more than 45 table games. The resort had seven restaurants in late 2005 and the hotel had 192 rooms. The resort opened on August 22, 2003, at a cost of $150 million. The resort has bus service from 20 cities. [edit] References ^ \"Chukchansi Gold Resort \u0026 Casino.\" 500 Nations. (retrieved 29 June 2010) [edit] External links Chukchansi Gold Resort \u0026 Casino website Hotel Online Coordinates: 37Â°12â²41â³N 119Â°41â²56â³Wï»¿ / ï»¿37.211254Â°N 119.698903Â°Wï»¿ / 37.211254; -119.698903 This article about a California building or structure is a stub. You can help Wikipedia by expanding it.v Â· d Â· e This article relating to a casino or gaming company is a stub. You can help Wikipedia by expanding it.v Â· d Â· e Retrieved from \"http://en.wikipedia.org/w/index.php?title\u003dChukchansi_Gold_Resort_%26_Casino\u0026oldid\u003d408967681\" Categories: Yokuts Casinos in California Native American casinos Buildings and structures in Madera County, California Visitor attractions in Madera County, California California building and structure stubs Casino stubs Personal tools Log in / create account Namespaces Article Talk Variants Views Read Edit View history Actions Search Navigation Main page Contents Featured content Current events Random article Donate to Wikipedia Interaction Help About Wikipedia Community portal Recent changes Contact Wikipedia Toolbox What links here Related changes Upload file Special pages Permanent link Cite this page Print/export Create a book Download as PDF Printable version This page was last modified on 20 January 2011 at 13:12. Text is available under the Creative Commons Attribution-ShareAlike License; additional terms may apply. See Terms of use for details. Wikipedia® is a registered trademark of the Wikimedia Foundation, Inc., a non-profit organization. Contact us Privacy policy About Wikipedia Disclaimers Mobile view",
        "fullyCanonicalizedContent": "chukchansi gold resort casino wikipedia free encyclopedia chukchansi gold resort casino from wikipedia free encyclopedia jump navig search chukchansi gold resort casino address 711 lucki lane coarsegold california 93614 room 192 total game space 56,000 sq ft 5,200 m2 notabl restaur california market buffet vintag steakhous deuc diner noodl bar asian bistro goldfield cafe bakeri firehous blue casino type land base owner picayun rancheria chukchansi indian websit http www.chukchansigold.com chukchansi gold resort casino locat just off state rout 41 coarsegold california between fresno yosemit nation park own oper picayun rancheria chukchansi indian 1 56,000 squar foot 5,200 m2 casino ha 1,800 slot machin well more than 45 tabl game resort had seven restaur late 2005 hotel had 192 room resort open august 22 2003 cost 150 million resort ha bu servic from 20 citi edit refer chukchansi gold resort casino 500 nation retriev 29 june 2010 edit extern link chukchansi gold resort casino websit hotel onlin coordin 37â 12â 41â n 119â 41â 56â wï ï 37.211254â n 119.698903â wï 37.211254 119.698903 articl about california build structur stub you can help wikipedia expand it.v â d â e articl relat casino game compani stub you can help wikipedia expand it.v â d â e retriev from http en.wikipedia.org w index.php titl chukchansi_gold_resort_ 26_casino oldid 408967681 categori yokut casino california nativ american casino build structur madera counti california visitor attract madera counti california california build structur stub casino stub person tool log creat account namespac articl talk variant view read edit view histori action search navig main page content featur content current event random articl donat wikipedia interact help about wikipedia commun portal recent chang contact wikipedia toolbox what link here relat chang upload file special page perman link cite page print export creat book download pdf printabl version page last modifi 20 januari 2011 13 12 text avail under creativ common attribut sharealik licens addit term mai appli see term us detail wikipedia ® regist trademark wikimedia foundat inc non profit organ contact u privaci polici about wikipedia disclaim mobil view",
        "crawlingTimestamp": "Wed, 11 Jan 2012 01:13:51 GMT",
        "url": "http://web.archive.org/web/20120111011351/http://en.wikipedia.org/wiki/Chukchansi_Gold_Resort_\u0026_Casino"
      }
    ]
  },
  {
    "_1": "clueweb09-enwp00-78-20657",
    "_2": [
      {
        "id": "\u003curn:uuid:2467d50d-4760-41fe-8a24-08901c243e2e\u003e",
        "content": "French Lick Township, Orange County, Indiana - Wikipedia, the free encyclopedia French Lick Township, Orange County, Indiana From Wikipedia, the free encyclopedia Jump to: navigation, search French Lick Township â  Township  â Location in Orange County Coordinates: 38Â°32â²40â³N 86Â°36â²41â³Wï»¿ / ï»¿38.54444Â°N 86.61139Â°Wï»¿ / 38.54444; -86.61139Coordinates: 38Â°32â²40â³N 86Â°36â²41â³Wï»¿ / ï»¿38.54444Â°N 86.61139Â°Wï»¿ / 38.54444; -86.61139 Country  United States State  Indiana County Orange Government  â¢ Type Indiana township Area  â¢ Total 54.05 sq mi (139.99 km2)  â¢ Land 53.98 sq mi (139.82 km2)  â¢ Water 0.07 sq mi (0.17 km2)  0.12% Elevation 482 ft (147 m) Population (2000)  â¢ Total 4,767  â¢ Density 88.3/sq mi (34.1/km2) ZIP codes 47432, 47454, 47469 GNIS feature ID 0453319 French Lick Township is one of ten townships in Orange County, Indiana, USA. As of the 2000 census, its population was 4,767.[1] Contents 1 Geography 1.1 Cities, towns, villages 1.2 Unincorporated towns 1.3 Cemeteries 1.4 Major highways 1.5 Airports and landing strips 2 School districts 3 Political districts 4 References 5 External links [edit] Geography According to the United States Census Bureau, French Lick Township covers an area of 54.05 square miles (140.0 km2); of this, 53.98 square miles (139.8 km2) (99.88 percent) is land and 0.07 square miles (0.18 km2) (0.12 percent) is water. [edit] Cities, towns, villages French Lick West Baden Springs [edit] Unincorporated towns Abydel at 38Â°34â²14â³N 86Â°33â²40â³Wï»¿ / ï»¿38.5706099Â°N 86.5611014Â°Wï»¿ / 38.5706099; -86.5611014 Prospect at 38Â°34â²41â³N 86Â°36â²51â³Wï»¿ / ï»¿38.5781095Â°N 86.6141585Â°Wï»¿ / 38.5781095; -86.6141585 Roland at 38Â°35â²43â³N 86Â°40â²55â³Wï»¿ / ï»¿38.595331Â°N 86.6819383Â°Wï»¿ / 38.595331; -86.6819383 (This list is based on USGS data and may include former settlements.) [edit] Cemeteries The township contains these three cemeteries: Mount Lebanon, Pythian and Sulphur Creek. [edit] Major highways U.S. Route 150 Indiana State Road 56 Indiana State Road 145 [edit] Airports and landing strips French Lick Municipal Airport [edit] School districts Springs Valley Community School Corporation [edit] Political districts Indiana\u0027s 9th congressional district State House District 62 State Senate District 48 [edit] References \"French Lick Township, Orange County, Indiana\". Geographic Names Information System, U.S. Geological Survey. http://geonames.usgs.gov/pls/gnispublic/f?p\u003dgnispq:3:::NO::P3_FID:0453319. Retrieved 2009-10-17.  United States Census Bureau 2008 TIGER/Line Shapefiles IndianaMap ^ United States Census Bureau American FactFinder [edit] External links Indiana Township Association United Township Association of Indiana City-Data.com page for French Lick Township Northwest Township Orangeville Township Lost River Township, Martin County Paoli Township    French Lick Township     Columbia Township, Dubois County Jackson Township Greenfield Township v d e Municipalities and communities of Orange County, Indiana, US County seat: Paoli Towns French Lick Orleans Paoli West Baden Springs Townships French Lick Greenfield Jackson Northeast Northwest Orangeville Orleans Paoli Southeast Stampers Creek Unincorporated communities Abydel Bacon Bonds Bromer Chambersburg Ethel Fargo Greenbrier Hindostan Leipsic Millersburg Newton Stewart Norton Orangeville Pearsontown Pine Valley Prospect Pumpkin Center Rego Roland Scarlet Syria Valeene Woodlawn Grove Youngs Creek Retrieved from \"http://en.wikipedia.org/w/index.php?title\u003dFrench_Lick_Township,_Orange_County,_Indiana\u0026oldid\u003d446869852\" Categories: Townships in Orange County, Indiana Personal tools Log in / create account Namespaces Article Talk Variants Views Read Edit View history Actions Search Navigation Main page Contents Featured content Current events Random article Donate to Wikipedia Interaction Help About Wikipedia Community portal Recent changes Contact Wikipedia Toolbox What links here Related changes Upload file Special pages Permanent link Cite this page Print/export Create a book Download as PDF Printable version Languages EspaÃ±ol This page was last modified on 26 August 2011 at 19:42. Text is available under the Creative Commons Attribution-ShareAlike License; additional terms may apply. See Terms of use for details. Wikipedia® is a registered trademark of the Wikimedia Foundation, Inc., a non-profit organization. Contact us Privacy policy About Wikipedia Disclaimers Mobile view",
        "fullyCanonicalizedContent": "french lick township orang counti indiana wikipedia free encyclopedia french lick township orang counti indiana from wikipedia free encyclopedia jump navig search french lick township â township â locat orang counti coordin 38â 32â 40â n 86â 36â 41â wï ï 38.54444â n 86.61139â wï 38.54444 86.61139coordin 38â 32â 40â n 86â 36â 41â wï ï 38.54444â n 86.61139â wï 38.54444 86.61139 countri unit state state indiana counti orang govern â type indiana township area â total 54.05 sq mi 139.99 km2 â land 53.98 sq mi 139.82 km2 â water 0.07 sq mi 0.17 km2 0.12 elev 482 ft 147 m popul 2000 â total 4,767 â densiti 88.3 sq mi 34.1 km2 zip code 47432 47454 47469 gni featur id 0453319 french lick township on ten township orang counti indiana usa 2000 censu it popul 4,767 1 content 1 geographi 1.1 citi town villag 1.2 unincorpor town 1.3 cemeteri 1.4 major highwai 1.5 airport land strip 2 school district 3 polit district 4 refer 5 extern link edit geographi accord unit state censu bureau french lick township cover area 54.05 squar mile 140.0 km2 53.98 squar mile 139.8 km2 99.88 percent land 0.07 squar mile 0.18 km2 0.12 percent water edit citi town villag french lick west baden spring edit unincorpor town abydel 38â 34â 14â n 86â 33â 40â wï ï 38.5706099â n 86.5611014â wï 38.5706099 86.5611014 prospect 38â 34â 41â n 86â 36â 51â wï ï 38.5781095â n 86.6141585â wï 38.5781095 86.6141585 roland 38â 35â 43â n 86â 40â 55â wï ï 38.595331â n 86.6819383â wï 38.595331 86.6819383 list base usg data mai includ former settlement edit cemeteri township contain three cemeteri mount lebanon pythian sulphur creek edit major highwai u. rout 150 indiana state road 56 indiana state road 145 edit airport land strip french lick municip airport edit school district spring vallei commun school corpor edit polit district indiana\u0027 9th congression district state hous district 62 state senat district 48 edit refer french lick township orang counti indiana geograph name inform system u. geolog survei http geonames.usgs.gov pl gnispubl f p gnispq 3 p3_fid 0453319 retriev 2009 10 17 unit state censu bureau 2008 tiger line shapefil indianamap unit state censu bureau american factfind edit extern link indiana township associ unit township associ indiana citi data.com page french lick township northwest township orangevil township lost river township martin counti paoli township french lick township columbia township duboi counti jackson township greenfield township v d e municip commun orang counti indiana u counti seat paoli town french lick orlean paoli west baden spring township french lick greenfield jackson northeast northwest orangevil orlean paoli southeast stamper creek unincorpor commun abydel bacon bond bromer chambersburg ethel fargo greenbrier hindostan leipsic millersburg newton stewart norton orangevil pearsontown pine vallei prospect pumpkin center rego roland scarlet syria valeen woodlawn grove young creek retriev from http en.wikipedia.org w index.php titl french_lick_township _orange_counti _indiana oldid 446869852 categori township orang counti indiana person tool log creat account namespac articl talk variant view read edit view histori action search navig main page content featur content current event random articl donat wikipedia interact help about wikipedia commun portal recent chang contact wikipedia toolbox what link here relat chang upload file special page perman link cite page print export creat book download pdf printabl version languag espaã ol page last modifi 26 august 2011 19 42 text avail under creativ common attribut sharealik licens addit term mai appli see term us detail wikipedia ® regist trademark wikimedia foundat inc non profit organ contact u privaci polici about wikipedia disclaim mobil view",
        "crawlingTimestamp": "Fri, 20 Jan 2012 14:06:31 GMT",
        "url": "http://web.archive.org/web/20120125060531/http://en.wikipedia.org:80/wiki/French_Lick_Township,_Orange_County,_Indiana"
      }
    ]
  }
]